areas/mysql/src/AzureMcp.MySql/Services/MySqlService.cs

```diff
@@ -0,0 +1,241 @@
+    }
+
+    private static string NormalizeServerName(string server)
+    {
+        if (!server.Contains('.'))
+        {
+            return server + ".mysql.database.azure.com";
+        }
+        return server;
+    }
+
+    public async Task<List<string>> ListDatabasesAsync(string subscriptionId, string resourceGroup, string user, string server)
+    {
+        var entraIdAccessToken = await GetEntraIdAccessTokenAsync();
+        var host = NormalizeServerName(server);
+        var connectionString = $"Server={host};Database=mysql;User ID={user};Password={entraIdAccessToken};SSL Mode=Required;";
+
+        await using var resource = await MySqlResource.CreateAsync(connectionString);
+        var query = "SHOW DATABASES;";
+        await using var command = new MySqlCommand(query, resource.Connection);
+        await using var reader = await command.ExecuteReaderAsync();
+        var dbs = new List<string>();
+        while (await reader.ReadAsync())
+        {
+            var dbName = reader.GetString(0);
+            // Filter out system databases
+            if (dbName != "information_schema" && dbName != "mysql" && dbName != "performance_schema" && dbName != "sys")
+            {
+                dbs.Add(dbName);
+            }
+        }
+        return dbs;
+    }
+
+    public async Task<List<string>> ExecuteQueryAsync(string subscriptionId, string resourceGroup, string user, string server, string database, string query)
+    {
+        var entraIdAccessToken = await GetEntraIdAccessTokenAsync();
+        var host = NormalizeServerName(server);
+        var connectionString = $"Server={host};Database={database};User ID={user};Password={entraIdAccessToken};SSL Mode=Required;";
+
+        await using var resource = await MySqlResource.CreateAsync(connectionString);
+        await using var command = new MySqlCommand(query, resource.Connection);
+        await using var reader = await command.ExecuteReaderAsync();
+
+        var rows = new List<string>();
+
+        var columnNames = Enumerable.Range(0, reader.FieldCount)
+                               .Select(reader.GetName)
+                               .ToArray();
+        rows.Add(string.Join(", ", columnNames));
+        while (await reader.ReadAsync())
+        {
+            var row = new List<string>();
+            for (int i = 0; i < reader.FieldCount; i++)
+            {
+                row.Add(reader[i]?.ToString() ?? "NULL");
+            }
+            rows.Add(string.Join(", ", row));
+        }
+        return rows;
+    }
+
+    public async Task<List<string>> GetTableSchemaAsync(string subscriptionId, string resourceGroup, string user, string server, string database, string table)
+    {
+        var entraIdAccessToken = await GetEntraIdAccessTokenAsync();
+        var host = NormalizeServerName(server);
+        var connectionString = $"Server={host};Database={database};User ID={user};Password={entraIdAccessToken};SSL Mode=Required;";
```

Comment:
If this string is used throughout, then refactor it out into a method that accepts optional params to build the string.